advanced/creating-a-plugin.md

```diff
@@ -1,43 +1,113 @@
 
-The build scripts that comes with the {{site.productname}} development package automatically builds all plugins including custom ones.
+{{site.productname}} does not require any special file structure or tooling apart from these requirements, so custom plugins can be developed using most frameworks and tools. 
 
-## Example of the plugin file structure
+### Yeoman Generator
 
+{{site.companyname}} maintains a [Yeoman generator]({{ site.baseurl }}/advanced/yeoman-generator/) to assist with creating plugins for {{site.productname}}. The Yeoman Generator will create the files and boilerplate code required for a custom plugin, and sets up some helpful commands. 
+
+## Registering a custom plugin with {{site.productname}}
+Register a custom plugin with {{site.productname}} using the PluginManager. `PluginManager.add()` takes a string for the plugin identifier and a function that contains the code for initializing the plugin. Optionally return an object containing a function that returns metadata which will be used in the list of plugins in the [Help plugin]({{ site.baseurl }}/plugins/opensource/help) dialog.
+
+The plugin identifier passed to `PluginManager.add()` is used by {{site.productname}} as an identifier string. It should be alphanumeric and can not contain spaces. Plugin identifiers should be unique. If multiple plugins have the same identifier, one will override the others. 
+
+The metadata object can contain the following values:
+
+- `name`: A string that contains the plugin's name, usually in a human-readable format.
+- `url`: A string that contains a URL, usually used to link to help documentation. 
+
+### Example: Plugin registration boilerplate
+
+```js
+tinymce.PluginManager.add('pluginId', function(editor, url) {
+  // add plugin code here
+
+  return {
+    getMetadata: function () {
+      return {
+        name: 'Custom plugin',
+        url: 'https://example.com/docs/customplugin'
+      };
+    }
+  }
+});
 ```
-/tinymce/plugins/example/plugin.js
-/tinymce/plugins/example/plugin.min.js
+
+## Using custom plugins with {{site.productname}}
+
+Custom plugins can be used by a {{site.productname}} instance by either:
+
+- **Using `external_plugins`**: Use the [external_plugins option]({{ site.baseurl }}/configure/integration-and-setup/#external_plugins) to specify the URL based location of the entry point file for the plugin.
+- **Copy code into `plugins` folder**: Copy the entry point file (and any other files) into the `plugins` folder of the distributed {{site.productname}} code.
+
+The plugin can then be used by including it in the list of plugins specified by the [`plugins` option]({{ site.baseurl }}/configure/integration-and-setup/#plugins).
```

Comment:
should this be a part of the point above, or is the example for `external_plugins` wrong (empty `plugins` list)?

Reply:
Ah, yes, should be moved up. Thanks!